apis/apiextensions/v1/composition_types.go

```diff
@@ -346,6 +347,7 @@ const (
 type Transform struct {
 
 	// Type of the transform to be run.
+	// +kubebuilder:validation:Enum=map;math;string;convert
```

Comment:
Just noting here while this is technically restricting the allowed value for this field at the API server, we are currently erroring out before creating _any_ composed managed resources, so this change should not break any current consumers of this functionality 👍